pkg/proxy/proxy.go

```diff
@@ -213,24 +213,42 @@ func (p *Proxy) CreateOrUpdateRedirect(l4 *policy.L4Filter, id string, localEndp
 	}
 
 	var to uint16
-	fixedProxyPort := false
-	listenerName := id
-	// Detect the need for a reusable listener here and override the defaults, e.g.:
-	// if <need reusable listener> {
-	//         listenerName = "cilium-xxxx-yyyy" // unique name for the reusable listener
-	//         to = <port> // listener port to use
-	//         fixedProxyPort = true // needed if `to` was overridden above
-	// }
-
-	if l4.L7Parser == policy.ParserTypeHTTP {
+	var listenerName string
+
+	// Detect the need for a reusable listener here and override the defaults:
+	switch l4.L7Parser {
+	case policy.ParserTypeHTTP:
 		if l4.Ingress {
 			listenerName = "cilium-http-ingress"
 			to = 6773
 		} else {
 			listenerName = "cilium-http-egress"
 			to = 6769
 		}
-		fixedProxyPort = true
+	case policy.ParserTypeKafka:
+		if l4.Ingress {
+			listenerName = "cilium-kafka-ingress"
+			to = 6774
```

Comment:
it will be useful if all of these ports were a constants on top of this file with documentation. Will we fail hard if we can't allocate all of those continuous ports?

Reply:
Proxy ports are now dynamically allocated.